libs/wingc/src/type_check.rs

```diff
@@ -2483,6 +2486,79 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	fn list_fields_on_init(&mut self, statements: &Scope) -> Vec<String> {
+		let mut fields: Vec<String> = Vec::new();
+		for stmt in &statements.statements {
+			match &stmt.kind {
+				StmtKind::Assignment { variable, value: _ } => match &variable {
+					Reference::InstanceMember { property, object: _ } => {
+						fields.push(property.name.clone());
+					}
+					_ => (),
+				},
```

Comment:
(optional) Another way to do this would be to create a visitor using the trait in `visit.rs`. Doing so might let you avoid having to write all of the cases for `If`/`While`/`ForLoop`/`TryCatch` since it will automatically do the recursion for those cases by default.

Reply:
Cool, I will try this